### ISSUE TITLE
Massively increased DNS requests

### ISSUE THREAD

**Issue by jchristianj** [dgreif/ring#741]
# Bug Report

### Describe the Bug
Since updating from 9.20.0 to 9.21.0 the number of DNS requests from home bridge has massively increased on my pi-hole.

### To Reproduce
Update to 9.21.0 and observe number of DNS queries.

### Expected behavior
Should not increase and be in the range of 9.20.0

### Screenshots/Logs

<img width="1244" alt="Bildschirmfoto 2021-08-16 um 14 44 30" src="https://user-images.githubusercontent.com/51285869/129566832-05ac3e03-143d-4bbe-b78c-fd6de4df2cd4.png">

Typo above in screenshot: Ring plugin is 9.21.0 (not 9.20.1)

<img width="480" alt="Bildschirmfoto 2021-08-16 um 15 02 44" src="https://user-images.githubusercontent.com/51285869/129568048-c69d8d38-9efd-424a-b2cd-5f12920121c2.png">

### Homebridge Ring Config
{
    "refreshToken": "XXSECRETXX",
    "platform": "Ring"
}
### Environment

 - OS: Raspbian GNU/Linux Buster (10)
 - Node.js: v14.16.1
 - NPM: v7.9.0
 - homebridge-ring: 9.21.0
 - homebridge: 1.3.4
 - hoobs: na

**Comment by dgreif** [comment]
I had to disable http2, which was being used to keep a connection open with the servers and cut down on dns requests.  Http2 just isn't ready for regular use in NodeJS yet.  It's been causing a ton of reliability issues over the last few months, so ultimately I had to turn it off.  I've tried a bunch of other DNS caching and keep-alive approaches in the past, and all of it comes with other issues.  In the end, the best approach is to just make simple requests and let DNS happen.  Sorry it's blowing up your pi-hole logs, but this is the most stable solution for the plugin right now.

**Comment by jchristianj** [comment]
Thanks for clarifying. Will try to filter these requests for the logs, or bypass the pi-hole for homebridge.

**Comment by Djelibeybi** [comment]
FYI, I "solved" this problem by running a local instance of `dnsmasq` for Homebridge with my Pi-hole as its upstream. This means the Pi-hole no longer sees all the DNS requests and cached responses. It only sees the far less regular request to refresh the cache (which is once every minute).

**Comment by tsightler** [comment]
@dgreif  One thing to note, it's not just about spamming logs, it's literally generating GB's of DNS traffic per day, even for fairly small setups, at least for customers who's systems don't cache DNS requests.  I had one customer report nearly 1TB in traffic (using all of their allotted bandwidth for the month) in just a few weeks.  I tested in my setup and it's about 35GB/day of DNS traffic if my HA instance uses a public DNS server like Google DNS.  I run a local DNS cache on my network so it's no big issue for me, and I think that's pretty common, but I guess there are some setups that don't cache local DNS requests and just forward every requests to the upstream DNS.

Another user reported that his internet connection started going on/offline just from starting the addon and this appears to be due to the sheer number of DNS requests going upstream.

What's not clear to me is why it's doing so many DNS lookups, even without caching.  I can't believe Node by default doesn't do basic, RFC complaint caching, but it seems it does not from what I've read.  Still, it just doesn't seem to make sense to me as the DNS traffic is far and away more than the actual HTTP traffic itself.

I'll probably get around it in ring-mqtt by running dnsmasq locally in the container, since 95% of users are now using the containerized version, but you might want to consider at least offering an option to enable request caching, even though I know it had some problems in the past.

**Comment by dgreif** [comment]
Do they have proof that the dns requests are the main source of data usage? That doesn't seem right. They can also tune the polling frequency to cut down on traffic quite a bit

**Comment by tsightler** [comment]
I actually tested it in my lab, if I set my DNS on my host to 8.8.8.8 on my host running even a simple script, it will easily generate 20-30MB/second of traffic even with just the alarm, no cameras.  It's crazy!  I don't really even understand how it can be that much, but as soon as I set DNS back to my local caching server the traffic to the internet dropped to nearly nothing.

I admittedly haven't captured a trace yet, but the observational evidence seemed strong (I let it run 24 hours with the DNS servers set to 8.8.8.8 and my firewall logged 35GB of traffic from the host running ring-mqtt, when I pointed it at my local DNS server it barely managed to get 200MB in 24 hours, it was basically a 200x difference in traffic.

At this point it's three reports in a week (not counting the one here) after years of never having anything like this reported and it seems trivially easy to reproduce in my lab when I don't point to my local caching DNS, similar to the user issues above, but bandwidth vs just hits.  

When I was playing with got I do remember seeing something like this one time, like got was timing out the requests in ms instead of seconds.  Maybe I need to dig into that a little since I believe you did still leave the timeout setting configured, but in my case I believe it was user error and I had actually set 20 ms for the timeout, where you clearly have 20000.  Is there any chance it's some type of non-critical call failing fast with like a 404 or something like that and just looping forever underneath?

Anyway, I will dig into it.  I wouldn't have brought it up because I thought there was no way it could possibly be ring-mqtt easting this much bandwidth, and certainly not DNS, but when I tested it in the lab, it seemed definitive, although it was just observation.  I will do the work and get the trace.

**Comment by tsightler** [comment]
Just to follow up on this, I really don't know what's going on.  I spent some time trying to get a trace with my firewall but once I finally sat down to do it I haven't been able to reproduce the issue again no matter what I try, so I have no idea what is going on.  Two of the three other users are also saying they now can't reproduce the issue.  The only thing I can think of is that somehow there was some request from Ring that was returning quickly, but with some status that was causing an immediate retry, although I see nothing in the code that could cause this to happen.

Basically, ignore the noise for now though.  If I can eventually reproduce it I will get the trace.

**Comment by dgreif** [comment]
Thanks for following up @tsightler.  You would think making HTTP request would be simple, but these DNS issues, keep cropping up 😅
